### PR TITLE
EUDEMO reactor starting point

### PR DESCRIPTION
**Pull request by CoronelBuendia**
* Port part of EUDEMO build for new architecture

* Remove design/examples directory

* Add TF coil builder to EUDEMO reactor

* Delete accidentally created __init__ file

* Add PF coil tools for eudemo

* Use 'read' mode for TF Coil design

This makes the reactor build much faster whilst I'm debugging other things!

* Add designer for EUDEMO PF coils

* Fix 'missing docstring' flake8 warning

* Run eudemo package tests in CI

* Add VV thermal shield to eudemo reactor

* Add template for EUDEMO PF coil builder

* Remove bad unit conversion in EUDEMO equilibrium

Co-authored-by: je-cook <81617086+je-cook@users.noreply.github.com>

* Remove another bad unit conversion in EUDEMO equil.

Co-authored-by: je-cook <81617086+je-cook@users.noreply.github.com>

* Fix TF coil parameter units in EUDEMO tests

Co-authored-by: je-cook <81617086+je-cook@users.noreply.github.com>

* Fix flake8 warnings in EUDEMO PF coils builder

* Add README to eudemo folder

* Remove reference to examples/design in pre-commit

* Fix path to data in EUDEMO ivc test

* Rename eudemo/tests -> eudemo/eudemo_tests

This gets around a module name clash (eudemo's tests and bluemira's tests) in pytest.

Also fix a couple of file paths in the EUDEMO tests

* Uncomment code in eudemo tools test

This does cause an error, but it's an error that needs fixing. Better to get the error as a reminder to fix it, than to forget about it.

* Fix path in .flake8 config

Co-authored-by: je-cook <81617086+je-cook@users.noreply.github.com>

## Linked Issues

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by CoronelBuendia** [comment]
@hsaunders1904 I will try to clean some of the code smells in a PR on this but ready soon

**Review by hsaunders1904 — APPROVED** [review]
All looks good to me. Happy for this to be merged once #1703 is in.

@je-cook, do you have any final thoughts? This is the feature branch, so it should all have been reviewed previously.

**Comment by CoronelBuendia** [comment]
@hsaunders1904 Don't know what's happened but `Divertor` and `VacuumVesselThermalShield` no longer exist:

```python
Traceback (most recent call last):
  File "eudemo/reactor.py", line 46, in <module>
    from bluemira.builders.divertor import Divertor, DivertorBuilder
ImportError: cannot import name 'Divertor' from 'bluemira.builders.divertor' (/home/matti/code/bluemira/bluemira/builders/divertor.py)

```
did something get lost along the way?

**Comment by je-cook** [comment]
> @hsaunders1904 Don't know what's happened but `Divertor` and `VacuumVesselThermalShield` no longer exist:
> 
> ```python
> Traceback (most recent call last):
>   File "eudemo/reactor.py", line 46, in <module>
>     from bluemira.builders.divertor import Divertor, DivertorBuilder
> ImportError: cannot import name 'Divertor' from 'bluemira.builders.divertor' (/home/matti/code/bluemira/bluemira/builders/divertor.py)
> ```
> 
> did something get lost along the way?

I think the Divertor ComponentManager was removed as it was empty in #1679 and will need to be created again as needed

File: eudemo/eudemo/reactor.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+class EUDEMO(Reactor):
+    """EUDEMO reactor definition."""
+
+    plasma: Plasma
+    vacuum_vessel: VacuumVessel
+    divertor: Divertor
+    blanket: Blanket
+    tf_coils: TFCoil
+    vv_thermal: VacuumVesselThermalShield
+
```
Comment (je-cook) [inline eudemo/eudemo/reactor.py:73]: Some of these component managers need to be recreated. I think just `Divertor` and ` VacuumVesselThermalShield`

**Review by je-cook — APPROVED** [review]
Looks good just the component managers comment

File: eudemo/eudemo/reactor.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    eq = run_designer(EquilibriumDesigner, params, build_config["Equilibrium"])
+
+    reactor.plasma = build_plasma(build_config.get("Plasma", {}), eq)
+
+    blanket_face, divertor_face, ivc_boundary = design_ivc(
+        params, build_config["IVC"], equilibrium=eq
+    )
+
+    # Do not add VV to the reactor, as the built shape errors when
+    # displayed (cannot build face from wire).
```
[the remark this answers is not in the document]
Comment (CoronelBuendia) [reply eudemo/eudemo/reactor.py:141]: Not even sure seeing as I need the new equilibrium stuff to check. The old eq is a tough job for the FW optimiser

**Comment by CoronelBuendia** [comment]
OK, so with #1712 in, I think this is good to go. It's a good enough start to start putting some larger PRs into, at least.

**Review by je-cook — APPROVED** [review]

**Pull request merged**